src/AutoRest.CSharp/Common/Output/Models/Types/SystemObjectType.cs

```diff
@@ -57,7 +57,7 @@ private ConstructorInfo GetCtor(string attributeType)
         {
             "ResourceIdentifier" => type,
             "SystemData" => type,
-            _ => ReferenceTypes.IsMgmtReferenceType (type) ? typeof(string) : null,
+            _ => null,
```

Comment:
This line is the reason that previously we have the explicit conversion - the `AzureLocation` and `ResourceType` are explicitly set to "SerializeAs: typeof(string)"

Reply:
I also doubt if we still need these hard-coded thing - but considering that this is not the main purpose of this PR, maybe we could do that later